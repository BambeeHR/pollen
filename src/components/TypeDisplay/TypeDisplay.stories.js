import Typography from '../../constants/Typography';
import TypeDisplay from './TypeDisplay.vue';
import TypeOverline from '../TypeOverline/TypeOverline.vue';

const { Display, Weights } = Typography;

export default {
  title: 'Typography/TypeDisplay',
  component: TypeDisplay,
};

export const Gallery = () => ({
  components: { TypeDisplay, TypeOverline },
  data() {
    return {
      variants: Object.values(Display),
      weights: [Weights.REGULAR, Weights.STRONG],
    };
  },
  template: `
    <div>
      <div v-for="variant in variants" :key="variant" :style="{ marginBottom: '48px' }">
        <TypeOverline tag="h1">{{ variant }}</TypeOverline>
        <TypeDisplay 
          v-for="weight in weights"
          :key="variant + '-' + weight"
          :variant="variant"
          :weight="weight"
          tag="div"
        >
          The quick brown fox jumps over the lazy dog.
        </TypeDisplay>
      </div>
    </div>
  `,
});
