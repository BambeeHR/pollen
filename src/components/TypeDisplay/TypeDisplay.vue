<template functional>
  <component
    :is="props.tag"
    :class="[
      `type-display-${props.variant}`,
      `type-display-${props.variant}--${props.weight}`,
      data.class,
      data.staticClass,
    ]"
    class="type-display"
    v-bind="$options.getAttributes({ data, parent })"
    v-on="listeners"
  >
    <slot />
  </component>
</template>

<script>
import Typography from '../../constants/Typography';
import getAttributes from '../../util/getAttributes';

const { Display, Weights } = Typography;

/**
 * `import { TypeDisplay } from '@bambee/pollen';`
 *
 * A styled block of text.
 */
export default {
  props: {
    /** The html tag to use for this component. */
    tag: {
      type: String,
      default: 'span',
    },
    /** One of `mega`, `giga`, `tera`, or `peta`. */
    variant: {
      type: String,
      default: Typography.Display.PETA,
      validator: (value) => Object.values(Display).includes(value),
    },
    /** One of `regular`, or `strong`. */
    weight: {
      type: String,
      default: Typography.Weights.STRONG,
      validator: (value) => [Weights.REGULAR, Weights.STRONG].includes(value),
    },
  },
  getAttributes,
};
</script>

<style>
@each $variant in mega, giga, tera, peta {
  .type-display-$(variant) {
    @apply font-display-$(variant);
  }

  .type-display-$(variant)--regular {
    @apply font-normal;
  }

  .type-display-$(variant)--strong {
    @apply font-bold;
  }
}
</style>
