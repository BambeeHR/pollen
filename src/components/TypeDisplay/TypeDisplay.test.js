import { mount } from '@vue/test-utils';
import TypeDisplay from './TypeDisplay.vue';

describe('TypeDisplay', () => {
  test('renders correctly', () => {
    const wrapper = mount(TypeDisplay, {
      slots: {
        default: 'Hello world!',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
