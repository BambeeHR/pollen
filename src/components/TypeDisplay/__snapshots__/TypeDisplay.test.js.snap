// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeDisplay renders correctly 1`] = `
<span
  class="type-display type-display-peta type-display-peta--strong"
>
  Hello world!
</span>
`;
